server/channels/app/desktop_login.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"net/http"
+
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+func (a *App) CreateDesktopToken(token string, createdAt int64) *model.AppError {
+	// Check if the token already exists in the database somehow
+	// If so return an error
+	_, getErr := a.Srv().Store().DesktopTokens().GetUserId(token, 0)
```

Comment:
So, it's incredibly unlikely that two users would try to create the same token twice but for the sake of argument let's assume that happens somehow (maybe they both crafted API calls with a `test_token` or something)

I would tend to think that table locking would prevent the same token from being inserted twice since there is a unique PK on the `DesktopToken` column. I think this extra check in there just kind of ensures that we don't have the token in here already somehow, and it makes it more explicit to check for that before we do anything to avoid an issue where the table wasn't locked or something.

I could see it being a bit of a performance hit to keep this, but I don't think it's significant and I don't want to prematurely optimize in this case.

Reply:
Ah, I missed that the token was the primary key. In that case, I'd still lean towards removing this since trying to insert a duplicate token is guaranteed to fail